AK/RedBlackTree.h

```diff
@@ -131,6 +131,25 @@ class BaseRedBlackTree {
         return candidate;
     }
 
+    static Node* find_smallest_not_below(Node* node, K key)
+    {
+        while (true) {
+            if (!node)
+                return node;
+
+            if (node->key < key && !node->right_child && !node->left_child)
+                return nullptr;
```

Comment:
There's no point to this check, it will be handled by the loop check implicitly.